rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/ContextMenu.java

```diff
@@ -225,20 +225,65 @@ public static Menu createContextMenuFromCompactView(RMController controller, Obj
 
     public void disableProviderItems(ContextMenu menu, String url) {
         LoginModel loginModel = LoginModel.getInstance();
-        if (!loginModel.userHasProviderPermissionForNodeSource(url)) {
+        boolean permissionForNode = loginModel.userHasProviderPermissionForNode(url);
+        if (!permissionForNode || !loginModel.userHasPermissionToLockNodes()) {
             disableLockMenuItem(menu);
+        }
+        if (!permissionForNode || !loginModel.userHasPermissionToUnLockNodes()) {
             disableUnlockMenuItem(menu);
+        }
+        if (!permissionForNode || !hasMethodPermissionToRemoveItem(menu)) {
             disableRemoveMenuItem(menu);
         }
     }
 
     public void disableAdminItems(ContextMenu menu, String url) {
         LoginModel loginModel = LoginModel.getInstance();
-        if (!loginModel.userHasAdminPermissionForNodeSource(url)) {
+        boolean permissionForNodeSource = loginModel.userHasAdminPermissionForNodeSource(url);
+        if (!permissionForNodeSource || !loginModel.userHasPermissionToDeployNodeSource()) {
             disableDeployItem(menu);
+        }
+        if (!permissionForNodeSource || !loginModel.userHasPermissionToUndeployNodeSource()) {
             disableUndeployItem(menu);
+        }
+        if (!permissionForNodeSource || !loginModel.userHasPermissionToEditNodeSource()) {
+            disableEditItem(menu);
+        }
+        if (!permissionForNodeSource && !loginModel.userHasPermissionToRemoveNodeSource()) {
             disableRemoveMenuItem(menu);
         }
+        if (!permissionForNodeSource && !hasMethodPermissionToEditItem(menu)) {
+            disableRemoveMenuItem(menu);
+        }
```

Comment:
Same comment as above, I think you should optimize the `!permissionForNodeSource` testing